src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -97,51 +163,84 @@ public IronPythonCompletionProvider()
                 try
                 {
-                    scope.Engine.CreateScriptSourceFromString("import clr\n", SourceCodeKind.Statements).Execute(scope);
-
                     var revitImports =
                         "clr.AddReference('RevitAPI')\nclr.AddReference('RevitAPIUI')\nfrom Autodesk.Revit.DB import *\nimport Autodesk\n";
 
-                    scope.Engine.CreateScriptSourceFromString(revitImports, SourceCodeKind.Statements).Execute(scope);
+                    engine.CreateScriptSourceFromString(revitImports, SourceCodeKind.Statements).Execute(scope);
+                    clrModules.Add("RevitAPI");
+                    clrModules.Add("RevitAPIUI");
                 }
                 catch
                 {
-                    Log("Failed to load Revit types for autocomplete.  Python autocomplete will not see Autodesk namespace types.");
+                    Log("Failed to load Revit types for autocomplete. Python autocomplete will not see Autodesk namespace types.");
                 }
             }
 
-            if (assemblies.Any(x => x.FullName.Contains("ProtoGeometry")))
+            // Determine if the ProtoGeometry is available in the given context
+            if (assemblies.Any(x => x.GetName().Name == "ProtoGeometry"))
             {
+                // Try to load ProtoGeometry Type for autocomplete
                 try
                 {
-                    scope.Engine.CreateScriptSourceFromString("import clr\n", SourceCodeKind.Statements).Execute(scope);
-
                     var libGImports =
-                        "import clr\nclr.AddReference('ProtoGeometry')\nfrom Autodesk.DesignScript.Geometry import *\n";
+                        "clr.AddReference('ProtoGeometry')\nfrom Autodesk.DesignScript.Geometry import *\n";
 
-                    scope.Engine.CreateScriptSourceFromString(libGImports, SourceCodeKind.Statements).Execute(scope);
+                    engine.CreateScriptSourceFromString(libGImports, SourceCodeKind.Statements).Execute(scope);
+                    clrModules.Add("ProtoGeometry");
                 }
                 catch (Exception e)
                 {
                     Log(e.ToString());
-                    Log("Failed to load ProtoGeometry types for autocomplete.  Python autocomplete will not see Autodesk namespace types.");
+                    Log("Failed to load ProtoGeometry types for autocomplete. Python autocomplete will not see Autodesk namespace types.");
                 }
             }
 
+            // Determine if the Python Standard Library is available in the given context
+            var pythonStdLib = pythonLibDir;
+
+            if (pythonStdLib != null && Directory.Exists(pythonStdLib))
+            {
+                // Try to load Python Standard Library Type for autocomplete
+                try
+                {
+                    var pyLibImports = String.Format("import sys\nsys.path.append(r'{0}')\n", pythonStdLib);
```

Comment:
does this still need to be done?

Reply:
yes it is required to load the autocomplete for the python std modules.  It is also being done for ProtoGeometry which is included in the template by default.